packages/@tailwindcss-upgrade/src/template/migrate.ts

```diff
@@ -0,0 +1,45 @@
+import fs from 'node:fs/promises'
+import path from 'node:path'
+import type { Candidate } from '../../../tailwindcss/src/candidate'
+import type { DesignSystem } from '../../../tailwindcss/src/design-system'
+import { extractCandidates, printCandidate, replaceCandidateInContent } from './candidates'
+import { migrateImportant } from './codemods/migrate-important'
+
+export type Migration = (candidate: Candidate) => Candidate | null
+
+export default async function migrateContents(
+  designSystem: DesignSystem,
+  contents: string,
+  migrations: Migration[] = [migrateImportant],
+): Promise<string> {
+  let candidates = await extractCandidates(designSystem, contents)
+
+  // Sort candidates by starting position desc
+  candidates.sort((a, z) => z.start - a.start)
```

Comment:
This is smart, I was wondering how you were planning to keep track of the position of candidates changing as you insert replacements 🧠